connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -168,6 +173,44 @@ public Connector newConnector(String connectorClassOrAlias) {
         return newPlugin(klass);
     }
 
+    public Converter newConverter(String className) throws ClassNotFoundException {
+        Class<? extends Converter> klass = pluginClass(
+                delegatingLoader,
+                className,
+                Converter.class
+        );
+        return newPlugin(klass);
+    }
+
+    public HeaderConverter newHeaderConverter(String className) throws ClassNotFoundException {
+        Class<? extends HeaderConverter> klass = pluginClass(
+                delegatingLoader,
+                className,
+                HeaderConverter.class
+        );
+        return newPlugin(klass);
+    }
+
+    @SuppressWarnings("rawtypes")
+    public Predicate newPredicate(String className) throws ClassNotFoundException {
```

Comment:
This should match the type-safe declaration for `newTransformation` IMO:

```suggestion
    @SuppressWarnings({"unchecked", "rawtypes"})
    public <R extends ConnectRecord<R>> Predicate<R> newPredicate(String className) throws ClassNotFoundException {
```

I think this is due to testing woes with the `AbstractHerderTest::testConnectorPluginConfig` test case; I've left a suggestion in that section for how to keep the type safety here with a small change there.